Plugin/NonBoltOrderPlugin.php

```diff
@@ -107,37 +106,35 @@ public function __construct(
     }
 
     /**
-     * @param Observer $observer
+     * @param OrderManagementInterface $orderManagementInterface
+     * @param Order
+     *
+     * @return Order
      */
-    public function execute(Observer $observer)
+    public function afterPlace(OrderManagementInterface $orderManagementInterface, $order)
```

Comment:
The code looks great and I see no way how we can have any issue with it.
But one more thing for discussion.

We add this code at a very critical moment when users create order.
I can imagine it takes something around 0.5 seconds (data collection + API request).

If someone add 0.5 second to bolt interaction with users I will be a little upset. :)
Can we find a more asynchronous way to collect the data? I guess it's not simple but maybe it worth doing.
Also, we can disable this by default.

What do you think @nickfiacco-bolt @matt-thomason @njfaries  ?

Reply:
This is a great point. What we can do is emit our own event using this plugin. Then add an observer which executed this logic asynchronously. Since we emit the event, we don't have to worry about inconsistent state. Will work on that today.

Reply:
On second look it seems observers are not async. We sort of need it to be synchronous anyway, since we need to ingest this data before it's used for tracking, etc

Reply:
Also-- it already disabled by default

Reply:
Another option would be to send the order information over in the Tracking observer, before sending the tracking info.

Reply:
So we need to discuss if we want to add this "asynchronous" things to backlog or not. I am not sure about it.

I don't see any simple solution.
1. we can save non bolt order orders when we handle bolt hooks, this is a good place for asynchronous work. But in this case, we can don't have actual info when a user wants to see their's account
2. we can get most part of data as (1) but also make an additional call from hail to plugin when any user wants to open account